bob/src/api/mod.rs

```diff
@@ -1,2 +1,1106 @@
+where
+    A: Authenticator,
+{
+    if !bob.auth().check_credentials(creds.into_credentials())?.has_rest_read() {
+        return Err(AuthError::PermissionDenied.into());
+    }
+    let backend = bob.grinder().backend().inner();
+    let (_, alien_disk_controller) = backend
+        .disk_controllers()
+        .ok_or_else(not_acceptable_backend)?;
+    let path = PathBuf::from(alien_disk_controller.disk().path());
+    let dir = create_directory(&path).await?;
+    Ok(Json(dir))
+}
+
+// GET /vdisks/:vdisk_id/replicas/local/dirs
+async fn get_local_replica_directories<A>(
+    bob: Extension<BobServer<A>>,
+    AxumPath(vdisk_id): AxumPath<u32>,
+    creds: CredentialsHolder<A>,
+) -> Result<Json<Vec<Dir>>, StatusExt>
+where
+    A: Authenticator,
+{
+    if !bob.auth().check_credentials(creds.into_credentials())?.has_rest_read() {
+        return Err(AuthError::PermissionDenied.into());
+    }
+    let vdisk: VDisk = get_vdisk_by_id(&bob, vdisk_id).ok_or_else(|| {
+        let msg = format!("VDisk {} not found", vdisk_id);
+        StatusExt::new(StatusCode::NOT_FOUND, false, msg)
+    })?;
+    let local_node_name = bob.grinder().backend().mapper().local_node_name();
+    let mut result = vec![];
+    for replica in vdisk
+        .replicas
+        .into_iter()
+        .filter(|r| r.node == local_node_name)
+    {
+        let path = PathBuf::from(replica.path);
+        let dir = create_directory(&path).await?;
+        result.push(dir);
+    }
+    Ok(Json(result))
+}
+
+fn create_directory(root_path: &Path) -> BoxFuture<Result<Dir, StatusExt>> {
+    async move {
+        let name = root_path
+            .file_name()
+            .and_then(|n| n.to_str())
+            .ok_or_else(|| internal(format!("failed to get filename for {:?}", root_path)))?;
+        let root_path_str = root_path
+            .to_str()
+            .ok_or_else(|| internal(format!("failed to get path for {:?}", root_path)))?;
+        let result = read_dir(root_path).await;
+        match result {
+            Ok(read_dir) => Ok(read_directory_children(read_dir, name, root_path_str).await),
+            Err(e) => Err(internal(format!("read path {:?} failed: {}", root_path, e))),
+        }
+    }
+    .boxed()
+}
+
+async fn read_directory_children(mut read_dir: ReadDir, name: &str, path: &str) -> Dir {
+    let mut children = Vec::new();
+    while let Ok(Some(entry)) = read_dir.next_entry().await {
+        let dir = create_directory(&entry.path()).await;
+        if let Ok(dir) = dir {
+            children.push(dir);
+        }
+    }
+    Dir {
+        name: name.to_string(),
+        path: path.to_string(),
+        children,
+    }
+}
+
+// GET /data/:key
+async fn get_data<A>(
+    bob: Extension<BobServer<A>>,
+    AxumPath(key): AxumPath<String>,
+    creds: CredentialsHolder<A>,
+) -> Result<impl IntoResponse, StatusExt>
+where
+    A: Authenticator,
+{
+    if !bob.auth().check_credentials(creds.into_credentials())?.has_rest_read() {
```

Comment:
For data access in REST API we should use `has_read`, `has_write`. Please, update `get_data`, `put_data`, `delete_data` here and in S3 API